interview-questions.md

```diff
@@ -6,62 +6,50 @@ First, without external resources. Challenge yourself to answer from memory as i
 
+Researched answer: Render is the most used lifecyle method and is required to have in a class component in React. Render is simply what tells React what to display onto the page. It displays the specified HTML code inside the specified HTML element.
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: JSX is a language we use in React it is mainly a combination of HTML and JavaScript. Even though it is a combination there are some syntax differences when using it. For example when you are assigning a method, variable, or pretty much anything instead of using "", we use {} instead. I have used JSX whenever I am inside a React application. I have specifically used the different syntax variations when I assigned an onClick to a function.
 
+Researched answer: JSX stands for Javascript XML, it allows devs to write HTML in React, it also makes it easier to add HTML in React. One main difference from HTML and JSX is that in JSX you can only return on thing and in order to do that we wrap everything we want to display in either a div or a React fragment.
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Yarn is a key word we use in out terminal before we create a new React application and when we are about to run jest testing. Although I dont quite know exactly what it means when we use it in terminal it does download node modules and is necessary to run testing and create a react application. I have used it several times but one for example is in the code-challenges.test.js we need to test our code and I had to use yarn to get that working.
 
+Researched answer: Yarn is a package manager for code. It allows devs to use and share code with other devs all over the world quickly, securely, and reliably.
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
-
-  Researched answer:
+Your answer: I dont think I have ever expierenced or used an anonymous function but a normal function is a reusable piece of code that we can invoke whenever wherever, significatly reducing the amount of typing the dev has to do. If I were to guess what an anonymous function is I would guess it is a function with no name.
 
+Researched answer: I was suprisingly right, an anonymous function is a function without a name so when the dev is creating the function instead of doing const function name and then arrow syntax, you just do const and arrow syntax. However with an anonymous function you cannot call it after creation because it does not have a name to call.
```

Comment:
A good example of an anonymous function is the call back function we pass to map(()=>{}) filter(()=>{})  or sort(()=>{})